traitsui/testing/tester/query.py

```diff
@@ -39,3 +39,13 @@ class DisplayedText:
     Implementations should return a ``str``.
     """
     pass
+
+
+class IsChecked:
+    """ An object representing an interaction to obtain whether a checkbox
+    is checked or empty.
+
+    Implementations should return something which evaluates to True if checked
+    and False if not.
```

Comment:
I'd suggest sticking with just `True` or `False`. Something which can be evaluated to true includes things like `[1]`, `"some-nonempty-string"`, ... we don't need to this flexible at this state. It is easier to go from strict requirement to a more lenient requirement later, whereas the reversed direction is not backward compatible.